viewing_party/main.py

```diff
@@ -0,0 +1,240 @@
+# Wave 1
+def create_movie(title, genre, rating):
+    # if bool(title) == True and bool(genre) == True and bool(rating) == True:
+    if title and genre and rating:
+        movie_dict = {
+            "title": title,
+            "genre": genre,
+            "rating": rating
+        }
+        return movie_dict
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    user_data = {
+        "watched": []
+    }
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data = {
+        "watchlist": []
+    }
+    user_data["watchlist"].append(movie)
+    return user_data
+
+'''
+If title is in a movie in the user's watchlist, remove that movie
+from watchlist and add that to watched.
+'''
+def watch_movie(user_data, title):
+
+    user_data_copy = user_data.copy()
+
+    for movie in user_data_copy["watchlist"]: # initial length: 2
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            return user_data
+            # break #Do I always need a break in a for-else loop? Or return will stop?
+                # else:
+                #     return user_data
+            # return use_data
+        # return use_data
+    else:
+        return user_data
+
+# Wave 2
+def get_watched_avg_rating(user_data):
+
+    # for v in user_data.values():
+    #     if v == []:
+    #         return 0.0
+    #     else:
+    #         for movie in v:
+    #             total_rating += movie["rating"]
+    #         avg = total_rating / len(v)
+    #         return avg
+
+    if user_data["watched"] == []:
+        return 0.0
+
+    total_rating = 0
+    for movie in user_data["watched"]:
+            total_rating += movie["rating"]
+            avg = total_rating / len(user_data["watched"])
```

Comment:
Each time this loop runs an avg is calculated, but only the last avg is the one that is needed.  Moving this line out of the for loop has the same effect.
```
for movie in user_data["watched"]:
            total_rating += movie["rating"]
avg = total_rating / len(user_data["watched"])
```